### PR TITLE
Remove some uses of riscv in tests

### PR DESCRIPTION
**Pull request by bjorn3**

### CODE REVIEW

**Comment by bnjbvr** [comment]
@bjorn3 What's the context of this PR, please?

**Comment by bjorn3** [comment]
I don't know if I will have enough of a riscv backend for the new backend framework before the old backend framework gets removed. This PR makes it possible to remove the old riscv backend together with the rest of the old backend framework without having to remove any uses of it outside the tests for the old backend framework.

File: cranelift/codegen/src/isa/mod.rs [hunk 1]
```diff
@@ -30,12 +29,12 @@
...
 //!
-//! match isa::lookup(triple!("riscv32")) {
+//! match isa::lookup(triple!("x86_64")) {
 //!     Err(_) => {
-//!         // The RISC-V target ISA is not available.
+//!         // The x86_64 target ISA is not available.
 //!     }
 //!     Ok(mut isa_builder) => {
-//!         isa_builder.set("supports_m", "on");
+//!         isa_builder.set("use_opcnt", "on");
```
Comment (bnjbvr) [inline cranelift/codegen/src/isa/mod.rs:37]: nit: popcnt

File: cranelift/reader/src/parser.rs [hunk 2]
```diff
@@ -3678,7 +3678,7 @@ mod tests {
     }
 
     #[test]
-    #[cfg(feature = "riscv")]
+    #[ignore] // FIXME this test is failing for some reason
```
Comment (bnjbvr) [inline cranelift/reader/src/parser.rs:3681]: Is it intended to mark this test as ignored, or did you plan to look into it? It's not obvious where the error is here; did you get a relevant backtrace or something that would help diagnose what's going on?

**Comment by bnjbvr** [comment]
> I don't know if I will have enough of a riscv backend for the new backend framework before the old backend framework gets removed. This PR makes it possible to remove the old riscv backend together with the rest of the old backend framework without having to remove any uses of it outside the tests for the old backend framework.

Makes sense!

File: cranelift/reader/src/parser.rs (hunk 2 above)
Comment (bjorn3) [reply cranelift/reader/src/parser.rs:3681]: It just fails to parse the test. I don't know why. This test was already effectively ignored previously as cranelift-reader doesn't have a riscv feature.

File: cranelift/codegen/src/isa/mod.rs [hunk 3]
```diff
@@ -30,12 +29,12 @@
...
 //!
-//! match isa::lookup(triple!("riscv32")) {
+//! match isa::lookup(triple!("x86_64")) {
 //!     Err(_) => {
-//!         // The RISC-V target ISA is not available.
+//!         // The x86_64 target ISA is not available.
 //!     }
 //!     Ok(mut isa_builder) => {
-//!         isa_builder.set("supports_m", "on");
+//!         isa_builder.set("use_pocnt", "on");
```
Comment (bnjbvr) [inline cranelift/codegen/src/isa/mod.rs:37]:
```suggestion
//!         isa_builder.set("use_popcnt", "on");
```

File: cranelift/reader/src/parser.rs (hunk 2 above)
Comment (bnjbvr) [inline cranelift/reader/src/parser.rs:3681]:
```suggestion

```

File: cranelift/reader/src/parser.rs [hunk 4]
```diff
@@ -3697,7 +3697,7 @@ mod tests {
 
         match parse_test(
             "set enable_float=false
-                          isa riscv
+                          isa x86_64
```
Comment (bnjbvr) [inline cranelift/reader/src/parser.rs:3700]:
```suggestion
                          target x86_64
```

File: cranelift/reader/src/parser.rs [hunk 5]
```diff
@@ -3707,7 +3707,7 @@ mod tests {
             IsaSpec::None(_) => panic!("Expected some ISA"),
             IsaSpec::Some(v) => {
                 assert_eq!(v.len(), 1);
-                assert_eq!(v[0].name(), "riscv");
+                assert_eq!(v[0].name(), "x86_64");
```
Comment (bnjbvr) [inline cranelift/reader/src/parser.rs:3710]:
```suggestion
                assert_eq!(v[0].name(), "x86");
```

**Review by bnjbvr — APPROVED** [review]
I've investigated the parser failure and fixed it.

File: cranelift/reader/src/parser.rs [hunk 6]
```diff
@@ -3707,7 +3706,7 @@ mod tests {
             IsaSpec::None(_) => panic!("Expected some ISA"),
             IsaSpec::Some(v) => {
                 assert_eq!(v.len(), 1);
-                assert_eq!(v[0].name(), "riscv");
+                assert_eq!(v[0].name(), "x86");
```
Comment (bnjbvr) [inline cranelift/reader/src/parser.rs:3709]:
```suggestion
                #[cfg(not(feature = "experimental_x64"))]
                assert_eq!(v[0].name(), "x86");
                #[cfg(feature = "experimental_x64")]
                assert_eq!(v[0].name(), "x64");
```

**Pull request merged**